tests/integration/testsuites/gateway/scenario.go

```diff
@@ -255,7 +291,8 @@ func (c *scenario) deleteVirtualService(name string) error {
 
 func (c *scenario) deleteAPIGatewayCR(name string) error {
 	res := schema.GroupVersionResource{Group: "operator.kyma-project.io", Version: "v1alpha1", Resource: "apigateways"}
-	err := c.k8sClient.Resource(res).Delete(context.Background(), name, v1.DeleteOptions{})
+	deletePolicy := metav1.DeletePropagationForeground
```

Comment:
I understand that foreground deletion will orphan all dependent resources. We should probably change the feature name accordingly since this seems generally unexpected from the step name.

Reply:
will remove it, since it may not be the behaviour we want within in scenario.

Reply:
we can run the tests in the tear-down hook as well. definitely I am against leaving orphaned objects, since there is a function to delete some stuff that could potentially block CR deletion. will run the hook also without this deletePolicy.